src/test/kotlin/com/github/codefabrikgmbh/scenarios/GivenTest.kt

```diff
@@ -1,20 +1,98 @@
 package com.github.codefabrikgmbh.scenarios
 
+import org.junit.Assert
 import org.junit.Test
 
-class `exception throwing scenario` : Scenario() {
-    init {
-        throw IllegalStateException("You created an invalided context!")
+class SpecialException(message: String, val specialValue: String) : RuntimeException(message)
+
+class `test scenario` : Scenario() {
+    var variable = ""
+
+    fun `given the variable is set to`(value: String) {
+        variable = value
+    }
+
+    fun `then the variable matches`(value: String) {
+        Assert.assertEquals(value, this.variable)
+    }
+
+    fun `when this throws an exception`(value: String) {
+        throw SpecialException("This is an exception!", value)
+    }
+
+    fun `then the user gets the message`(e: SpecialException, value: String) {
```

Comment:
The word 'user' has no meaning in this context and implies a specific scenario that has not been modelled here. 

Just write `then a exception has a message` or maybe `then the exception has a message` but i prefer the first version as it complies with our convention to only use THE when referring to something within the scenario.